projects/ipdg_stokes/assemble/test/matrix_assembly_tests.cc

```diff
@@ -0,0 +1,221 @@
+#include <gtest/gtest.h>
+
+#include <piecewise_const_element_matrix_provider.h>
+
+#include <lf/assemble/assemble.h>
+#include <lf/assemble/coomatrix.h>
+#include <lf/assemble/dofhandler.h>
+#include <lf/base/base.h>
+#include <lf/base/forward_range.h>
+#include <lf/geometry/geometry.h>
+#include <lf/geometry/tria_o1.h>
+#include <lf/mesh/hybrid2d/mesh_factory.h>
+#include <lf/mesh/utils/all_codim_mesh_data_set.h>
+#include <lf/mesh/utils/utils.h>
+
+#include <cmath>
+
+TEST(projects_ipdg_stokes_assembly, piecewise_const_matrix_assembler_test) {
+  // Build a mesh containing only the reference triangle
+  const auto trig = lf::base::RefEl::kTria();
+  const auto vertices = trig.NodeCoords();
+  std::unique_ptr<lf::geometry::Geometry> geom =
+      std::make_unique<lf::geometry::TriaO1>(vertices);
+  const lf::base::ForwardRange<const lf::mesh::MeshFactory::size_type> nodes(
+      {0, 1, 2});
+  lf::mesh::hybrid2d::MeshFactory factory(2);
+  factory.AddPoint(vertices.col(0));
+  factory.AddPoint(vertices.col(1));
+  factory.AddPoint(vertices.col(2));
+  factory.AddEntity(trig, nodes, std::move(geom));
+  const auto mesh = factory.Build();
+  const auto element = mesh->EntityByIndex(0, 0);
+  // Compute the element matrix for the triangle
+  const lf::mesh::utils::AllCodimMeshDataSet<bool> boundary(
+      mesh, false);  // Assume that this triangle does not touch the boundary
+  const double sigma = 1;
+  const auto elem_mat_provider =
+      projects::ipdg_stokes::assemble::PiecewiseConstElementMatrixProvider(
+          sigma, boundary);
+  const Eigen::MatrixXd Ae = elem_mat_provider.Eval(*element);
+  // Construct the analytically computed element matrix
+  Eigen::MatrixXd Ae_anal(6, 6);
+  const double r = std::sqrt(2);
+  const double b = 1. / std::sqrt(2);
+  const double s = 1. / (2 * sigma);
+  // clang-format off
+  Ae_anal <<  0,  0,  0, -1,  r, -1,
+              0,  0,  0,  0, -b,  1,
+	      0,  0,  0,  1, -b,  0,
+	     -1,  0,  1, -s,  0,  0,
+	      r, -b, -b,  0, -s,  0,
+	     -1,  1,  0,  0,  0, -s;
+  // clang-format on
+  // Assert that the two matrices are approximately equal
+  ASSERT_EQ(Ae.rows(), Ae_anal.rows());
+  ASSERT_EQ(Ae.cols(), Ae_anal.cols());
+  for (int i = 0; i < Ae_anal.rows(); ++i) {
+    for (int j = 0; j < Ae_anal.cols(); ++j) {
+      EXPECT_DOUBLE_EQ(Ae(i, j), Ae_anal(i, j))
+          << "mismatch in entry (" << i << ", " << j << ")";
+    }
+  }
```

Comment:
I actually deliberately chose this method of comparing the two matrices exactly because the error messages are much more useful for debugging the element matrix assembler. I can however change it if it would be more consistent with the other code in the repository.